addons/mrp/models/mrp_production.py

```diff
@@ -1036,9 +1037,18 @@ def _set_qty_producing(self):
         for move in (self.move_raw_ids | self.move_finished_ids.filtered(lambda m: m.product_id != self.product_id)):
             if move._should_bypass_set_qty_producing() or not move.product_uom:
                 continue
+
             new_qty = float_round((self.qty_producing - self.qty_produced) * move.unit_factor, precision_rounding=move.product_uom.rounding)
+            if self.use_auto_consume_components_lots and move.has_tracking in ('lot', 'serial'):
+                if float_compare(move.reserved_availability, 0, precision_rounding=move.product_uom.rounding) <= 0:
+                    continue
+                else:
+                    new_qty = min(new_qty, move.reserved_availability)
+
             move.move_line_ids.filtered(lambda ml: ml.state not in ('done', 'cancel')).qty_done = 0
             move.move_line_ids = move._set_quantity_done_prepare_vals(new_qty)
+            if move.has_tracking == 'serial':
```

Comment:
I tried locally without this `if` and the mrp passed. Could you make a new runbot build without it to check if everything works fine ? Thanks